### PR TITLE
Get raw gene count tables from either Salmon and RSEM analysis 

### PR DESCRIPTION
**Pull request by qbicStefanC**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/qbic-pipelines/rnadeseq/tree/master/.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by qbicStefanC** [comment]
Hi @WackerO 
I created this to get (back) the raw counts from the Salmon/RSEM paths inside rnadeseq. I think the changes in the file are abvious. I have tested it with a real project dataset (QTRRI) and with the test profile and made sure other files or results are unchanged. I think that part is fine . What I am still a bit unsure about is the extraction of the raw counts either how i did it with : assay(cds) or alternatively with counts(cds). In above mentioned project it did not make a difference but maybe once could test this again. Thanks,S

**Comment by WackerO** [comment]
Hello @qbicStefanC, thanks for the PR! I think the assay() function is fine.

It would be nice if you could update the test YMLs (https://github.com/qbic-pipelines/rnadeseq/blob/master/tests/test_star_salmon.yml and https://github.com/qbic-pipelines/rnadeseq/blob/master/tests/test_star_rsem.yml) by adding entries for the new matrices that are output; you just need to run those two tests (i.e. nextflow run main.nf -profile cfc,test_star_rsem and test_star_salmon) and generate the md5sums for the matrices.

Also, please update the https://github.com/qbic-pipelines/rnadeseq/blob/dev/CHANGELOG.md by copy-pasting an older entry to the top and modifying the line :)

**Comment by WackerO** [comment]
Hey @qbicStefanC, thanks for the tests! The changelog looks good, can you resolve the conflict so we can see if the new tests succeed?

**Comment by qbicStefanC** [comment]
resolved the conflict on the changelog...

File: assets/RNAseq_report.Rmd [hunk 1]
```diff
@@ -592,6 +592,10 @@ if (params$input_type %in% c("featurecounts", "smrnaseq")) {
 
         #dds from SummarizedExperiment <se>, then run DESeq
         cds <- DESeqDataSet(se, design = as.formula(eval(parse(text=as.character(design[[1]])))))
+        # get raw counts
```
Comment (WackerO) [inline assets/RNAseq_report.Rmd:595]:
```suggestion
        # get raw counts
```

File: assets/RNAseq_report.Rmd [hunk 2]
```diff
@@ -625,6 +629,10 @@ if (params$input_type %in% c("featurecounts", "smrnaseq")) {
         coldata$combfactor <- metadata$combfactor
         rownames(coldata) <- qbicCodes
         cds <- DESeqDataSetFromTximport(txi=txi.salmon, colData =coldata, design = eval(parse(text=as.character(design[[1]]))))
+        # get raw counts
```
Comment (WackerO) [inline assets/RNAseq_report.Rmd:632]:
```suggestion
        # get raw counts
```

**Comment by WackerO** [comment]
Thanks for the work, Stefan! I just removed some trailing whitespaces so prettier won't complain. The tests should now all run (except for linting), so after that is done, I'll override the protection and merge :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -32,7 +33,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Removed
 
 - [#247](https://github.com/qbic-pipelines/rnadeseq/pull/247) Removed hard-coded comment about usage of gprofiler databases KEGG and REAC in the report
-
+
```
Comment (WackerO) [inline CHANGELOG.md:36]:
```suggestion

```

**Comment by WackerO** [comment]
The remaining tests fail due to an old template, will override and merge

**Review by WackerO — APPROVED** [review]

**Pull request merged**